metrics/exporter.go

```diff
@@ -161,13 +162,13 @@ func UpdateExporter(ctx context.Context, ops ExporterOptions, logger *zap.Sugare
 		flushGivenExporter(curMetricsExporter)
 		e, f, err := newMetricsExporter(newConfig, logger)
 		if err != nil {
-			logger.Errorw("Failed to update a new metrics exporter based on metric config", newConfig, zap.Error(err))
+			logger.Errorw("Failed to update a new metrics exporter based on metric config", "config", newConfig, "error", err)
```

Comment:
zap.String("config", newConfig), zap.Error(err)? Same below?

Reply:
Ugh... I want to replace this with a non-sugared logger. It looks like [`With`](https://godoc.org/go.uber.org/zap#SugaredLogger.With) takes *both* Field objects and key-value pairs. Unfortunately, this ended up with the key-value format where the value was a `zap.Field`.

WDYT?

Reply:
Well you do use sugared.
You don't need `with`, just `logger.Error("...", zap.Error(err))` on the desugared one

Reply:
I switched this to put all the `Field` objects first so it's not possible to log things as "string" => Field, which was what the old code was doing.

Using `logger.Error()` means taking the error out of a separate JSON field, which makes it harder (for example) to filter all the logs for just the ones with errors in them.

Reply:
Yeah, I know. That's why when I joined I was surprised we;re using this crap (zap that is) :)
But that's the zap philosophy — use jq to parse :)